packages/lib/src/vis/shared/ZoomSelectionMesh.tsx

```diff
@@ -0,0 +1,63 @@
+import { useThree } from '@react-three/fiber';
+import type { Vector2 } from 'three';
+
+import type { ModifierKey } from '../models';
+import { useAxisSystemContext } from './AxisSystemContext';
+import SelectionMesh from './SelectionMesh';
+import SelectionRect from './SelectionRect';
+import { useMoveCameraTo } from './hooks';
+
+interface Props {
+  modifierKey?: ModifierKey;
+}
+
+function ZoomSelectionMesh(props: Props) {
+  const { modifierKey } = props;
+
+  const { dataToWorld } = useAxisSystemContext();
+  const moveCameraTo = useMoveCameraTo();
+
+  const { width, height } = useThree((state) => state.size);
+  const camera = useThree((state) => state.camera);
+
+  const onSelection = (startPoint: Vector2, endPoint: Vector2) => {
+    if (startPoint.x === endPoint.x && startPoint.y === endPoint.y) {
+      return;
+    }
+
+    const worldStartPoint = dataToWorld(startPoint);
+    const worldEndPoint = dataToWorld(endPoint);
+    const rectWidth = Math.abs(worldEndPoint.x - worldStartPoint.x);
+    const rectHeight = Math.abs(worldEndPoint.y - worldStartPoint.y);
+
+    // Change scale first so that moveCameraTo computes the updated camera bounds
+    camera.scale.set(rectWidth / width, rectHeight / height, 1);
+    camera.updateProjectionMatrix();
+    camera.updateMatrixWorld();
+
+    const midPoint = worldEndPoint
+      .clone()
+      .sub(worldStartPoint)
+      .divideScalar(2)
+      .add(worldStartPoint);
+
+    moveCameraTo(midPoint.x, midPoint.y);
```

Comment:
The zooming works in two steps:
- First, we need to set the camera scale so that the dimensions of the rectangle corresponds to the dimensions of the FOV. This is a simple proportional rule, for example for x :

| scale | FOV |
 | -- | -- |
 | 1 | `width` |
| x | `rectWidth` |

=> `x = rectWidth / width`

- Then, the camera must be moved to the center of the rectangle (called `midPoint` but `rectCenter` might be a better name). For this, I make use of the `moveCameraTo` utility that moves the camera in the vis bounds. As explained by the comment, the scale must be computed first to update camera matrices so that `moveCameraTo` computes the correct camera bounds.

Reply:
I agree with `rectCenter` or `selectionCenter` (and `selectionWidth/Height`)